app/services/unified_chat_service.py

```
@@ -1565,34 +1487,19 @@ def _pct(value: Any) -> Optional[str]:
-                                    expected_return = expected_return.strip()
-                                    if expected_return.endswith('%'):
-                                        expected_return = expected_return[:-1].strip()
-                                    expected_return = float(expected_return)
-
-                                # Convert to numeric and normalize
-                                expected_return = float(expected_return)
-                                # If abs(value) <= 1, treat as fraction; otherwise as percentage
-                                if abs(expected_return) <= 1.0:
-                                    expected_return *= 100
-
+                            formatted_expected = _format_percentage(expected_return)
```

Ruff (0.13.1)

Undefined name `_format_percentage`

(F821)